### ISSUE TITLE
Sitewide - Heading structure

### ISSUE THREAD

**Issue by gggrant** [GCTC-NTGC/TalentCloud#666]
# Description
Homepage:
- Need an invisible H1 "GC Talent Cloud" when a logo acts as the main heading.
- "How it Works" / "Our Team" should be H2
- "Contribute" etc should be H3

Other pages:
- Bump the headings up one level. i.e. "Browse Jobs" should be H1
- Job title should be H2, and so forth

Styles don't need to change, only structure.

**Comment by joshdrink** [comment]
@gggrant Is it cool if this issue is left ongoing for a while? Restructuring every page is going to take some thought, particularly because a lot of nested partials use headings at different points on a page. For now I've pushed the recommended fixes to the homepage in PR #670.
